src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DataTypesTest.java

```diff
@@ -1946,6 +1944,17 @@ public void testDateTimeOffsetValueOfOffsetDateTime() throws Exception {
         assertEquals(expected, DateTimeOffset.valueOf(roundDown).getOffsetDateTime());
     }
 
+    private OffsetDateTime prepareOffsetDateTime() {
+        OffsetDateTime value = OffsetDateTime.now().truncatedTo(ChronoUnit.MILLIS);
+        int offsetSeconds = value.getOffset().getTotalSeconds();
+        offsetSeconds += offsetSeconds < 0 ? 3600 : -3600;
+        return value.withOffsetSameLocal(ZoneOffset.ofTotalSeconds(offsetSeconds));
+    }
+
+    private LocalDateTime convertOffsetDateTimeToLocalDateTime(OffsetDateTime value) {
+        return value.atZoneSameInstant(ZoneId.systemDefault()).toLocalDateTime();
+    }
+

```

Comment:
I'm not opposed to splitting the tests but refactoring into these methods seem unnecessary since they are only called once, why not just include them in the tests themselves?

Reply:
Thanks for the feedback! The decision to refactor was driven by enhancing test readability and making future maintenance easier. This approach will improve the clarity of our test cases and reduces debugging rerun time when debugging for specific scenario. When tests are more granular and focused, identifying and addressing issues becomes faster, leading to more efficient debugging cycles.

Even though the methods are called only once now, this structure allows us to easily expand or modify tests with minimal changes. The reduction in debugging rerun time and the enhancement in clarity are pivotal, as they ensure that our test suite remains robust, facilitating quicker modifications.

Reply:
I don't think it makes it any easier to read. If this changes in the future it can be changed then for now there is no need to do unnecessary refactoring and extra method calls

Reply:
Thanks for suggestion! Your perspective on the overall development picture is very helpful. I agree that maintainability should be balanced with immediate clarity. Let's keep the tests straightforward as you've advised.